### ISSUE TITLE
Build from Git repository fails on Windows

### ISSUE THREAD

**Issue by clin1234** [lapce/lapce#3412]
```
~/src $ cargo install --locked --git https://github.com/lapce/lapce lapce
    Updating git repository `https://github.com/lapce/lapce`
  Installing lapce v0.4.0 (https://github.com/lapce/lapce#e00d5ca7)
    Updating crates.io index
   Compiling proc-macro2 v1.0.79
   Compiling unicode-ident v1.0.4
   Compiling serde v1.0.204
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
   Compiling once_cell v1.19.0
   Compiling autocfg v1.1.0
   Compiling log v0.4.20
   Compiling windows_x86_64_msvc v0.48.5
   Compiling memchr v2.7.1
   Compiling jobserver v0.1.31
   Compiling zerocopy v0.7.32
   Compiling allocator-api2 v0.2.16
   Compiling thiserror v1.0.63
   Compiling syn v1.0.109
   Compiling scopeguard v1.1.0
   Compiling winapi v0.3.9
   Compiling equivalent v1.0.1
   Compiling pin-project-lite v0.2.13
   Compiling getrandom v0.2.7
   Compiling ahash v0.8.11
   Compiling windows_x86_64_msvc v0.52.0
   Compiling windows_x86_64_msvc v0.36.1
   Compiling crossbeam-utils v0.8.19
   Compiling cc v1.0.99
   Compiling anyhow v1.0.69
   Compiling lock_api v0.4.9
   Compiling libc v0.2.155
   Compiling parking_lot_core v0.9.3
   Compiling crc32fast v1.3.2
   Compiling semver v1.0.23
   Compiling heck v0.4.1
   Compiling itoa v1.0.3
   Compiling rand_core v0.6.4
   Compiling tinyvec_macros v0.1.0
   Compiling tracing-core v0.1.30
   Compiling serde_json v1.0.122
   Compiling windows-targets v0.48.5
   Compiling rustc-hash v1.1.0
   Compiling pkg-config v0.3.25
   Compiling futures-core v0.3.30
   Compiling vcpkg v0.2.15
   Compiling tinyvec v1.6.0
   Compiling windows-sys v0.48.0
   Compiling typenum v1.15.0
   Compiling windows-targets v0.52.0
   Compiling windows-sys v0.36.1
   Compiling unicode-width v0.1.13
   Compiling ppv-lite86 v0.2.16
   Compiling arrayvec v0.7.2
   Compiling unicode-bidi v0.3.15
   Compiling cfg_aliases v0.1.1
   Compiling hashbrown v0.14.5
   Compiling either v1.12.0
   Compiling libz-sys v1.1.8
   Compiling futures-io v0.3.30
   Compiling windows-sys v0.52.0
   Compiling quote v1.0.35
   Compiling target-lexicon v0.12.11
   Compiling cranelift-isle v0.101.4
   Compiling aho-corasick v1.1.3
   Compiling rand_chacha v0.3.1
   Compiling slab v0.4.9
   Compiling cranelift-codegen-shared v0.101.4
   Compiling base64 v0.21.7
   Compiling ryu v1.0.11
   Compiling byteorder v1.4.3
   Compiling regex-syntax v0.8.2
   Compiling futures-sink v0.3.30
   Compiling syn v2.0.57
   Compiling rand v0.8.5
   Compiling cranelift-codegen-meta v0.101.4
   Compiling generic-array v0.14.6
   Compiling leb128 v0.2.5
   Compiling windows_x86_64_msvc v0.42.2
   Compiling io-lifetimes v2.0.2
   Compiling futures-channel v0.3.30
   Compiling libssh2-sys v0.3.0
   Compiling memoffset v0.6.5
   Compiling regex-automata v0.4.6
   Compiling wasm-encoder v0.35.0
   Compiling slotmap v1.0.7
   Compiling slice-group-by v0.3.0
   Compiling unicode-segmentation v1.11.0
   Compiling pin-utils v0.1.0
   Compiling simd-adler32 v0.3.7
   Compiling cranelift-codegen v0.101.4
   Compiling bitflags v1.3.2
   Compiling futures-task v0.3.30
   Compiling serde_derive v1.0.204
   Compiling thiserror-impl v1.0.63
   Compiling bytemuck_derive v1.5.0
   Compiling wasmtime-versioned-export-macros v14.0.2
   Compiling tracing-attributes v0.1.23
   Compiling futures-macro v0.3.30
   Compiling arbitrary v1.3.1
   Compiling zstd-sys v2.0.1+zstd.1.5.2
   Compiling crossbeam-epoch v0.9.10
   Compiling io-extras v0.18.0
   Compiling rustix v0.38.32
   Compiling bytes v1.5.0
   Compiling winapi-util v0.1.5
   Compiling cranelift-control v0.101.4
   Compiling bumpalo v3.14.0
   Compiling adler v1.0.2
   Compiling errno v0.3.8
   Compiling windows-targets v0.42.2
   Compiling crossbeam-channel v0.5.12
   Compiling miniz_oxide v0.7.1
   Compiling libloading v0.8.1
   Compiling encoding_rs v0.8.34
   Compiling wasmtime-component-util v14.0.2
   Compiling khronos_api v3.1.0
   Compiling ipnet v2.5.0
   Compiling percent-encoding v2.3.1
   Compiling ambient-authority v0.0.2
   Compiling cap-primitives v2.0.0
   Compiling flate2 v1.0.30
   Compiling fs-set-times v0.20.0
   Compiling mio v0.8.11
   Compiling wasmtime-fiber v14.0.2
   Compiling socket2 v0.5.7
   Compiling form_urlencoded v1.2.1
   Compiling tokio-macros v2.3.0
   Compiling unicode-normalization v0.1.22
   Compiling ittapi-sys v0.3.5
   Compiling num-traits v0.2.15
   Compiling memoffset v0.9.0
   Compiling maybe-owned v0.3.4
   Compiling zstd-safe v5.0.2+zstd.1.5.2
   Compiling rayon-core v1.12.1
   Compiling futures-util v0.3.30
   Compiling bytemuck v1.16.3
   Compiling libm v0.2.7
   Compiling memmap2 v0.9.0
   Compiling num_cpus v1.13.1
   Compiling ttf-parser v0.20.0
   Compiling dirs-sys v0.3.7
   Compiling idna v0.5.0
   Compiling crossbeam-deque v0.8.2
   Compiling indexmap v2.2.6
   Compiling wasmtime-runtime v14.0.2
   Compiling libgit2-sys v0.17.0+1.8.1
   Compiling proc-macro2-diagnostics v0.10.1
   Compiling cpufeatures v0.2.5
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling cpp_demangle v0.3.5
   Compiling cap-std v2.0.0
   Compiling id-arena v2.2.1
   Compiling raw-window-handle v0.6.0
   Compiling async-trait v0.1.74
   Compiling unicode-xid v0.2.4
   Compiling xml-rs v0.8.19
   Compiling digest v0.10.7
   Compiling tracing v0.1.37
   Compiling wast v35.0.2
   Compiling gl_generator v0.14.0
   Compiling dirs v4.0.0
   Compiling windows-sys v0.45.0
   Compiling dirs-sys-next v0.1.2
   Compiling regex v1.10.5
   Compiling com_macros_support v0.6.0
   Compiling itertools v0.10.5
   Compiling wasmtime-asm-macros v14.0.2
   Compiling unicode-bidi-mirroring v0.1.0
   Compiling sha2 v0.10.8
   Compiling uuid v1.5.0
   Compiling unicode-ccc v0.1.2
   Compiling wasmtime-wmemcheck v14.0.2
   Compiling unicode-script v0.5.5
   Compiling paste v1.0.9
   Compiling stable_deref_trait v1.2.0
   Compiling yansi v1.0.1
   Compiling wasmtime-cache v14.0.2
   Compiling float-cmp v0.9.0
   Compiling rustc-demangle v0.1.21
   Compiling sptr v0.3.2
   Compiling unicode-properties v0.1.1
   Compiling com_macros v0.6.0
   Compiling debugid v0.8.0
   Compiling itertools v0.12.1
   Compiling directories-next v2.0.0
   Compiling shellexpand v2.1.2
   Compiling strict-num v0.1.1
   Compiling witx v0.9.1
   Compiling glutin_wgl_sys v0.6.0
   Compiling rayon v1.10.0
   Compiling ittapi v0.3.5
   Compiling font-types v0.5.2
   Compiling fontdb v0.16.2
   Compiling wasmtime-jit-icache-coherence v14.0.2
   Compiling termcolor v1.4.1
   Compiling fdeflate v0.3.0
   Compiling wast v66.0.2
   Compiling fxhash v0.2.1
   Compiling naga v22.0.0
   Compiling concurrent-queue v2.2.0
   Compiling windows-core v0.51.1
   Compiling fnv v1.0.7
   Compiling static_assertions v1.1.0
   Compiling ash v0.38.0+1.3.281
   Compiling arrayref v0.3.7
   Compiling bit-vec v0.7.0
   Compiling khronos-egl v6.0.0
   Compiling png v0.17.10
   Compiling read-fonts v0.19.0
   Compiling http v0.2.8
   Compiling codespan-reporting v0.11.1
   Compiling wiggle-generate v14.0.2
   Compiling tiny-skia-path v0.11.4
   Compiling ouroboros_macro v0.18.3
   Compiling windows v0.51.1
   Compiling bit-set v0.6.0
   Compiling com v0.6.0
   Compiling atomic v0.6.0
   Compiling wgpu-hal v22.0.0
   Compiling tracing-core v0.2.0 (https://github.com/tokio-rs/tracing?rev=908cc432a5994f6e17c8f36e13c217dc40085704#908cc432)
   Compiling lazy_static v1.4.0
   Compiling wasi-common v14.0.2
   Compiling wat v1.0.77
   Compiling xxhash-rust v0.8.7
   Compiling widestring v1.0.2
   Compiling yazi v0.1.6
   Compiling siphasher v0.3.11
   Compiling hexf-parse v0.2.1
   Compiling httparse v1.8.0
   Compiling rustversion v1.0.14
   Compiling cap-fs-ext v2.0.0
   Compiling aliasable v0.1.3
   Compiling native-tls v0.2.10
   Compiling zeno v0.2.2
   Compiling seahash v4.1.0
   Compiling presser v0.3.1
   Compiling wit-parser v0.12.1
   Compiling gpu-allocator v0.26.0
   Compiling wiggle-macro v14.0.2
   Compiling sys-locale v0.2.4
   Compiling schannel v0.1.23
   Compiling kurbo v0.9.5
   Compiling wgpu-core v22.0.0
   Compiling simplecss v0.2.1
   Compiling profiling v1.0.10
   Compiling unicode-linebreak v0.1.5
   Compiling imagesize v0.12.0
   Compiling xmlwriter v0.1.0
   Compiling try-lock v0.2.3
   Compiling swash v0.1.15
   Compiling pico-args v0.5.0
   Compiling futures-executor v0.3.30
   Compiling system-interface v0.26.0
   Compiling glow v0.13.1
   Compiling weezl v0.1.7
   Compiling futures v0.3.30
   Compiling ouroboros v0.18.3
   Compiling fastrand v2.0.1
   Compiling parking v2.2.0
   Compiling data-url v0.3.1
   Compiling bytecount v0.6.3
   Compiling range-alloc v0.1.3
   Compiling renderdoc-sys v1.1.0
   Compiling rangemap v1.3.0
   Compiling color_quant v1.1.0
   Compiling unicode-vo v0.1.0
   Compiling roxmltree v0.19.0
   Compiling zune-core v0.4.12
   Compiling litrs v0.4.1
   Compiling event-listener v5.2.0
   Compiling want v0.3.0
   Compiling url v2.5.2
   Compiling gif v0.12.0
   Compiling wasmtime-wit-bindgen v14.0.2
   Compiling tiny-skia v0.11.4
   Compiling http-body v0.4.5
   Compiling zune-jpeg v0.4.11
   Compiling image-webp v0.1.1
   Compiling wasmtime-component-macro v14.0.2
   Compiling document-features v0.2.10
   Compiling rgb v0.8.36
   Compiling cap-rand v2.0.0
   Compiling fd-lock v4.0.0
   Compiling bstr v1.7.0
   Compiling serde_repr v0.1.9
   Compiling wgpu v22.0.0
   Compiling tree-sitter v0.22.6
   Compiling cfg_aliases v0.2.0
   Compiling httpdate v1.0.2
   Compiling smallvec v1.13.2
   Compiling bitflags v2.6.0
   Compiling cranelift-entity v0.101.4
   Compiling gimli v0.28.0
   Compiling wasmparser v0.115.0
   Compiling object v0.32.1
   Compiling regalloc2 v0.9.3 (https://github.com/bytecodealliance/regalloc2?rev=5d79e12d0a93b10fc181f4da409b4671dd365228#5d79e12d)
   Compiling parking_lot v0.12.3
   Compiling winx v0.36.2
   Compiling memfd v0.6.4
   Compiling toml v0.5.9
   Compiling bincode v1.3.3
   Compiling tokio v1.38.0
   Compiling cranelift-bforest v0.101.4
   Compiling wasmtime-types v14.0.2
   Compiling wasmprinter v0.2.70
   Compiling addr2line v0.21.0
   Compiling rustybuzz v0.12.1
   Compiling wasmtime-jit-debug v14.0.2
   Compiling triomphe v0.1.9
   Compiling fxprof-processed-profile v0.6.0
   Compiling wg v0.9.1
   Compiling kurbo v0.10.4
   Compiling kurbo v0.11.0
   Compiling gpu-descriptor-types v0.2.0
   Compiling gpu-alloc-types v0.3.0
   Compiling spirv v0.3.0+sdk-1.3.268.0
   Compiling serde_bytes v0.11.14
   Compiling gpu-descriptor v0.3.0
   Compiling stretto v0.8.4
   Compiling svgtypes v0.14.0
   Compiling gpu-alloc v0.6.0
error[E0046]: not all trait items implemented, missing: `change_time`
   --> C:/Users/ךינשגכהד\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cap-primitives-2.0.0\src\windows\fs\metadata_ext.rs:151:1
    |
151 | impl std::os::windows::fs::MetadataExt for MetadataExt {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `change_time` in implementation
    |
    = help: implement the missing item: `fn change_time(&self) -> Option<u64> { todo!() }`

error[E0046]: not all trait items implemented, missing: `change_time`
   --> C:/Users/ךינשגכהד\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cap-primitives-2.0.0\src\fs\metadata.rs:406:1
    |
406 | impl std::os::windows::fs::MetadataExt for Metadata {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `change_time` in implementation
    |
    = help: implement the missing item: `fn change_time(&self) -> Option<u64> { todo!() }`

For more information about this error, try `rustc --explain E0046`.
error: could not compile `cap-primitives` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `lapce v0.4.0 (https://github.com/lapce/lapce#e00d5ca7)`, intermediate artifacts can be found at `C:\Users\0BAC~1\AppData\Local\Temp\cargo-install4CKSYs`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

**Comment by jm-observer** [comment]
it is work: `cargo run --profile fastdev --bin lapce`

**Comment by sunfishcode** [comment]
I've now released cap-std 2.0.2, which I expect fixes this.

**Comment by panekj** [comment]
We don't support unstable Rust versions, so I'd recommend switching to stable before trying anything (https://github.com/lapce/lapce/blob/master/rust-toolchain.toml)
